crates/cli-support/src/js/mod.rs

```diff
@@ -842,7 +847,43 @@ impl<'a> Context<'a> {
         Ok(())
     }
 
-    fn gen_init(&mut self, module_name: &str, needs_manual_start: bool) -> String {
+    fn ts_for_init_fn(has_memory: bool) -> &'static str {
+        macro_rules! initial_ts {
+            () => ("
+                \n\
+                /**\n\
+                * If `module_or_path` is {RequestInfo}, makes a request and\n\
+                * for everything else, calls `WebAssembly.instantiate` directly.\n\
+                *\n\
+                * @param {RequestInfo | BufferSource | WebAssembly.Module} module_or_path\n\
+            ");
+        }
+        if has_memory {
+            concat!(initial_ts!(), "\
+            * @param {WebAssembly.Memory} maybe_memory\n\
+            *\n\
+            * @returns {Promise<WebAssembly.ResultObject | WebAssembly.Instance>}\n\
+            */\n\
+            export function init (\
+                module_or_path: RequestInfo | BufferSource | WebAssembly.Module,\
```

Comment:
Could this be deduplicated in the case `!has_memory` to avoid the duplication here?